packages/dds/sequence/src/test/intervalCollection.spec.ts

```diff
@@ -1731,5 +1759,166 @@ describe("SharedString interval collections", () => {
 				assert.equal(collection.detachIndex(mockIntervalIndex), false);
 			});
 		});
+
+		describe("support querying intervals with endpoint in a range", () => {
+			let helpers: IIntervalHelpers<Interval>;
+			let endInRangeIndex: EndInRangeIndex<Interval>;
+			let compareFn;
+			let results;
+
+			beforeEach(() => {
+				helpers = {
+					compareEnds: (a: Interval, b: Interval) => a.end - b.end,
+					create: createInterval,
+				};
+				endInRangeIndex = new EndInRangeIndex(helpers);
+				// sort the query result by the interval endpoint value
+				compareFn = (a: Interval, b: Interval) => {
+					if (a.end === b.end) {
+						return a.start - b.start;
+					}
+					return a.end - b.end;
+				};
+			});
+
+			it("can not return result if the input does not meet specified requirement", () => {
+				results = endInRangeIndex.findIntervalsWithEndInRange(1, 1);
+				assert.equal(
+					results.length,
+					0,
+					"Should not return anything once the index is empty",
+				);
+
+				endInRangeIndex.add(new Interval(1, 1));
+				endInRangeIndex.add(new Interval(1, 3));
+
+				results = endInRangeIndex.findIntervalsWithEndInRange(2, 1);
+				assert.equal(
+					results.length,
+					0,
+					"The start value should not be lower than the end value",
+				);
+
+				results = endInRangeIndex.findIntervalsWithEndInRange(0, 1);
+				assert.equal(results.length, 0, "The start value should be larger than 0");
+			});
+
+			it("can find correct results after adding multiple intervals", () => {
+				endInRangeIndex.add(new Interval(1, 1));
+				endInRangeIndex.add(new Interval(1, 3));
+
+				results = endInRangeIndex.findIntervalsWithEndInRange(1, 1);
+				assertPlainNumberInterval(results[0], { start: 1, end: 1 });
+
+				results = endInRangeIndex.findIntervalsWithEndInRange(1, 3);
+				assertPlainNumberInterval(results[0], { start: 1, end: 1 });
```

Comment:
might be worth writing a matches function that takes in an array. seems like it would be less verbose